watertap/unit_models/tests/test_cstr.py

```diff
@@ -71,214 +57,195 @@
-                    * sum(
-                        sapon.fs.unit.inlet.conc_mol_comp[0, j]
-                        for j in sapon.fs.properties.component_list
-                    )
-                    - sapon.fs.unit.outlet.flow_vol[0]
-                    * sum(
-                        sapon.fs.unit.outlet.conc_mol_comp[0, j]
-                        for j in sapon.fs.properties.component_list
-                    )
-                )
+        ] = value(
+            m.fs.unit.inlet.flow_vol[0]
+            * sum(
+                m.fs.unit.inlet.conc_mol_comp[0, j]
+                for j in m.fs.properties.component_list
             )
-            <= 1e-6
         )
 
-        assert pytest.approx(3904.51, abs=1e-2) == value(
-            sapon.fs.unit.control_volume.heat_of_reaction[0]
-        )
-        assert (
-            abs(
-                value(
-                    (
-                        sapon.fs.unit.inlet.flow_vol[0]
-                        * sapon.fs.properties.dens_mol
-                        * sapon.fs.properties.cp_mol
-                        * (
-                            sapon.fs.unit.inlet.temperature[0]
-                            - sapon.fs.properties.temperature_ref
-                        )
+        self.unit_solutions[
+            (
+                (
+                    m.fs.unit.outlet.flow_vol[0]
+                    * m.fs.properties.dens_mol
+                    * m.fs.properties.cp_mol
+                    * (
+                        m.fs.unit.outlet.temperature[0]
+                        - m.fs.properties.temperature_ref
                     )
-                    - (
-                        sapon.fs.unit.outlet.flow_vol[0]
-                        * sapon.fs.properties.dens_mol
-                        * sapon.fs.properties.cp_mol
-                        * (
-                            sapon.fs.unit.outlet.temperature[0]
-                            - sapon.fs.properties.temperature_ref
-                        )
-                    )
-                    + sapon.fs.unit.control_volume.heat_of_reaction[0]
                 )
+                - m.fs.unit.control_volume.heat_of_reaction[0]
             )
-            <= 1e-3
+        ] = value(
+            m.fs.unit.inlet.flow_vol[0]
+            * m.fs.properties.dens_mol
+            * m.fs.properties.cp_mol
+            * (m.fs.unit.inlet.temperature[0] - m.fs.properties.temperature_ref)
         )
 
-    @pytest.mark.ui
-    @pytest.mark.unit
-    def test_get_performance_contents(self, sapon):
-        perf_dict = sapon.fs.unit._get_performance_contents()
-
-        assert perf_dict == {
-            "vars": {
-                "Volume": sapon.fs.unit.volume[0],
-                "Heat Duty": sapon.fs.unit.heat_duty[0],
-                "Pressure Change": sapon.fs.unit.deltaP[0],
-            }
-        }
+        return m
 
 
 class TestInitializers:
     @pytest.fixture
     def model(self):
+        m = build()
```

Comment:
This class seems mostly unchanged. Are you able to make use of `self.unit_solutions` here instead of the assert statements?

Reply:
we are testing the initialization methods here not the solutions

Reply:
If I were to channel my "inner Tim", I'd say is testing the initialization methods necessary? If not, we should delete it. If so, we should consider doing something similar for other unit models or adding it to the harness (which might not be feasible/a good idea)

Reply:
To add my 2 cents: the `Initializers` are not part of WaterTAP and are tested as part of IDAES, so the question becomes twofold:

1. do you trust IDAES not to break things on you, and
2. what do you do if these tests start failing?

From a software engineering point of view, the "best" answer to this would be to add this case study to the IDAES backward compatibility tests. That way IDAES will see immediately if they break the tests, rather than having to wait for WaterTAP to discover and report it.